vsts-fullbuild-redhat-vm/azuredeploy.json

```diff
@@ -0,0 +1,205 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vstsAccountURL": {
+      "type": "string",
+      "metadata": {
+        "description": "Team Services Account URL (e.g. https://myaccount.visualstudio.com)"
+      }
+    },
+    "vstsPAT": {
+      "type": "string",
+      "metadata": {
+        "description": "Team Services PAT for user with Agent Pools (read,manage) permissions"
+      }
+    },
+    "vstsPoolName": {
+      "type": "string",
+      "defaultValue": "Default",
+      "metadata": {
+        "description": "Team Services Agent Pool Name"
+      }
+    },
+    "vstsAgentName": {
+      "type": "string",
+      "defaultValue": "redhat-build-full",
+      "metadata": {
+        "description": "Team Services Agent Name"
+      }
+    },    
+    "adminUserName": {
+      "type": "string",
+      "defaultValue": "vstsbuild",
+      "metadata": {
+        "description": "Linux VM User Account Name"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring"
+    },
+    "dnsLabelPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "DNS Label for the Public IP. Must be lowercase. It should match with the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$ or it will raise an error."
+      }
+    }
+  },
+  "variables": {
+    "imagePublisher": "redhat",
+    "imageOffer": "rhel",
+    "imageSKU":"7.2",
+    "baseName": "[uniquestring(parameters('dnsLabelPrefix'),resourceGroup().id,uniquestring(deployment().name))]",
```

Comment:
using deployment().name will mean this is not idempotent (in some workflows), is that intentional?